I'm compiling with MSVC and getting the following errors, Is it related to the previous part?

```
Build started at 16:26...
Waiting for background operations to complete...
1>------ Build started: Project: LLVMCore, Configuration: Release x64 ------
1>  Metadata.cpp
1>include\llvm\ADT\FoldingSet.h(450,27): error C2440: 'static_cast': cannot convert from 'llvm::FoldingSetNode *' to 'llvm::AttributeImpl *'
1>  (compiling source file '../../../llvm/lib/IR/Metadata.cpp')
1>      include\llvm\ADT\FoldingSet.h(450,27):
1>      Types pointed to are unrelated; conversion requires reinterpret_cast, C-style cast or parenthesized function-style cast
1>      include\llvm\ADT\FoldingSet.h(450,27):
1>      the template instantiation context (the oldest one first) is
1>          lib\IR\LLVMContextImpl.h(1605,29):
1>          see reference to class template instantiation 'llvm::FoldingSetImpl<llvm::AttributeImpl,llvm::FoldingSetTrait<T,void>>' being compiled
1>          with
1>          [
1>              T=llvm::AttributeImpl
1>          ]
1>include\llvm\ADT\FoldingSet.h(461,33): error C2440: 'static_cast': cannot convert from 'llvm::FoldingSetNode *' to 'llvm::AttributeImpl *'
1>  (compiling source file '../../../llvm/lib/IR/Metadata.cpp')
1>      include\llvm\ADT\FoldingSet.h(461,33):
1>      Types pointed to are unrelated; conversion requires reinterpret_cast, C-style cast or parenthesized function-style cast
1>include\llvm\ADT\FoldingSet.h(471,38): error C2440: 'static_cast': cannot convert from 'llvm::FoldingSetNode *' to 'llvm::AttributeImpl *'
1>  (compiling source file '../../../llvm/lib/IR/Metadata.cpp')
1>      include\llvm\ADT\FoldingSet.h(471,38):
1>      Types pointed to are unrelated; conversion requires reinterpret_cast, C-style cast or parenthesized function-style cast
1>include\llvm\ADT\FoldingSet.h(446,58): error C2440: 'initializing': cannot convert from 'initializer list' to 'const llvm::FoldingSetBase::FoldingSetInfo'
1>  (compiling source file '../../../llvm/lib/IR/Metadata.cpp')
1>      include\llvm\ADT\FoldingSet.h(446,58):
1>      Invalid aggregate initialization
1>          include\llvm\ADT\FoldingSet.h(466,7):
1>          'initializing': cannot convert from 'llvm::FoldingSetImpl<llvm::AttributeImpl,llvm::FoldingSetTrait<T,void>>::<lambda_9>' to 'bool (__cdecl *)(const llvm::FoldingSetBase *,llvm::FoldingSetBase::Node *,const llvm::FoldingSetNodeID &,unsigned int,llvm::FoldingSetNodeID &)'
1>          with
1>          [
1>              T=llvm::AttributeImpl
1>          ]
1>              include\llvm\ADT\FoldingSet.h(466,7):
1>              No user-defined-conversion operator available that can perform this conversion, or the operator cannot be called
1>          include\llvm\ADT\FoldingSet.h(476,7):
1>          'initializing': cannot convert from 'llvm::FoldingSetImpl<llvm::AttributeImpl,llvm::FoldingSetTrait<T,void>>::<lambda_10>' to 'unsigned int (__cdecl *)(const llvm::FoldingSetBase *,llvm::FoldingSetBase::Node *,llvm::FoldingSetNodeID &)'
1>          with
1>          [
1>              T=llvm::AttributeImpl
1>          ]
1>              include\llvm\ADT\FoldingSet.h(476,7):
1>              No user-defined-conversion operator available that can perform this conversion, or the operator cannot be called
1>include\llvm\ADT\FoldingSet.h(450,27): error C2440: 'static_cast': cannot convert from 'llvm::FoldingSetNode *' to 'llvm::AttributeListImpl *'
1>  (compiling source file '../../../llvm/lib/IR/Metadata.cpp')
1>      include\llvm\ADT\FoldingSet.h(450,27):
1>      Types pointed to are unrelated; conversion requires reinterpret_cast, C-style cast or parenthesized function-style cast
1>      include\llvm\ADT\FoldingSet.h(450,27):
1>      the template instantiation context (the oldest one first) is
1>          lib\IR\LLVMContextImpl.h(1606,33):
1>          see reference to class template instantiation 'llvm::FoldingSetImpl<llvm::AttributeListImpl,llvm::FoldingSetTrait<T,void>>' being compiled
1>          with
1>          [
1>              T=llvm::AttributeListImpl
1>          ]
1>include\llvm\ADT\FoldingSet.h(461,33): error C2440: 'static_cast': cannot convert from 'llvm::FoldingSetNode *' to 'llvm::AttributeListImpl *'
1>  (compiling source file '../../../llvm/lib/IR/Metadata.cpp')
1>      include\llvm\ADT\FoldingSet.h(461,33):
1>      Types pointed to are unrelated; conversion requires reinterpret_cast, C-style cast or parenthesized function-style cast
1>include\llvm\ADT\FoldingSet.h(471,38): error C2440: 'static_cast': cannot convert from 'llvm::FoldingSetNode *' to 'llvm::AttributeListImpl *'
1>  (compiling source file '../../../llvm/lib/IR/Metadata.cpp')
1>      include\llvm\ADT\FoldingSet.h(471,38):
1>      Types pointed to are unrelated; conversion requires reinterpret_cast, C-style cast or parenthesized function-style cast
1>include\llvm\ADT\FoldingSet.h(450,27): error C2440: 'static_cast': cannot convert from 'llvm::FoldingSetNode *' to 'llvm::AttributeSetNode *'
1>  (compiling source file '../../../llvm/lib/IR/Metadata.cpp')
1>      include\llvm\ADT\FoldingSet.h(450,27):
1>      Types pointed to are unrelated; conversion requires reinterpret_cast, C-style cast or parenthesized function-style cast
1>      include\llvm\ADT\FoldingSet.h(450,27):
1>      the template instantiation context (the oldest one first) is
1>          lib\IR\LLVMContextImpl.h(1607,32):
1>          see reference to class template instantiation 'llvm::FoldingSetImpl<llvm::AttributeSetNode,llvm::FoldingSetTrait<T,void>>' being compiled
1>          with
1>          [
1>              T=llvm::AttributeSetNode
1>          ]
1>include\llvm\ADT\FoldingSet.h(461,33): error C2440: 'static_cast': cannot convert from 'llvm::FoldingSetNode *' to 'llvm::AttributeSetNode *'
1>  (compiling source file '../../../llvm/lib/IR/Metadata.cpp')
1>      include\llvm\ADT\FoldingSet.h(461,33):
1>      Types pointed to are unrelated; conversion requires reinterpret_cast, C-style cast or parenthesized function-style cast
1>include\llvm\ADT\FoldingSet.h(471,38): error C2440: 'static_cast': cannot convert from 'llvm::FoldingSetNode *' to 'llvm::AttributeSetNode *'
1>  (compiling source file '../../../llvm/lib/IR/Metadata.cpp')
1>      include\llvm\ADT\FoldingSet.h(471,38):
1>      Types pointed to are unrelated; conversion requires reinterpret_cast, C-style cast or parenthesized function-style cast
========== Build: 0 succeeded, 1 failed, 0 up-to-date, 0 skipped ==========
========== Build completed at 16:26 and took 04.540 seconds ==========

```

--- a/llvm/include/llvm/ADT/FoldingSet.h
+++ b/llvm/include/llvm/ADT/FoldingSet.h
@@ -31,7 +31,7 @@
 
 namespace llvm {
 
-/// This folding set used for two purposes:
+/// This folding set is used for two purposes:
 ///   1. Given information about a node we want to create, look up the unique
 ///      instance of the node in the set.  If the node already exists, return
 ///      it, otherwise return the bucket it should be inserted into.
@@ -97,7 +97,8 @@ namespace llvm {
 /// 3) If you get a NULL result from FindNodeOrInsertPos then you can insert a
 /// new node with InsertNode;
 ///
-///    MyFoldingSet.InsertNode(M, InsertPoint);
+///    MyNode *N = new MyNode(Name, Value);
+///    MyFoldingSet.InsertNode(N, InsertPoint);
 ///
 /// 4) Finally, if you want to remove a node from the folding set call;
 ///
@@ -380,7 +381,7 @@ protected:
   /// was removed or false if the node was not in the folding set.
   LLVM_ABI bool RemoveNode(Node *N);
 
-  /// If there is an existing simple Node exactly equal to the node \p N,
+  /// If there is an existing node exactly equal to the node \p N,
   /// return it.  Otherwise, insert \p N and return it instead.
   LLVM_ABI Node *GetOrInsertNode(Node *N, const FoldingSetInfo &Info);
 
@@ -464,7 +465,7 @@ public:
   /// was removed or false if the node was not in the folding set.
   bool RemoveNode(T *N) { return FoldingSetBase::RemoveNode(N); }
 
-  /// If there is an existing simple Node exactly equal to the specified node,
+  /// If there is an existing node exactly equal to the specified node,
   /// return it.  Otherwise, insert 'N' and return it instead.
   T *GetOrInsertNode(T *N) {
     return static_cast<T *>(
@@ -641,7 +642,7 @@ public:
     return Set.FindNodeOrInsertPos(ID, InsertPos);
   }
 
-  /// If there is an existing simple Node exactly equal to the specified node,
+  /// If there is an existing node exactly equal to the specified node,
   /// return it.  Otherwise, insert 'N' and return it instead.
   T *GetOrInsertNode(T *N) {
     T *Result = Set.GetOrInsertNode(N);

--- a/llvm/lib/Support/FoldingSet.cpp
+++ b/llvm/lib/Support/FoldingSet.cpp
@@ -15,7 +15,6 @@
 #include "llvm/ADT/STLExtras.h"
 #include "llvm/ADT/StringRef.h"
 #include "llvm/Support/Allocator.h"
-#include "llvm/Support/ErrorHandling.h"
 #include "llvm/Support/MathExtras.h"
 #include "llvm/Support/SwapByteOrder.h"
 #include <cassert>
@@ -164,7 +163,7 @@ static void **GetBucketFor(unsigned Hash, void **Buckets, unsigned NumBuckets) {
   return Buckets + BucketNum;
 }
 
-/// AllocateBuckets - Allocated initialized bucket memory.
+/// AllocateBuckets - Allocate initialized bucket memory.
 static void **AllocateBuckets(unsigned NumBuckets) {
   void **Buckets =
       static_cast<void **>(safe_calloc(NumBuckets + 1, sizeof(void *)));
@@ -225,7 +224,6 @@ void FoldingSetBase::GrowBucketCount(unsigned NewBucketCount,
 
   // Clear out new buckets.
   Buckets = AllocateBuckets(NewBucketCount);
-  // Set NumBuckets only if allocation of new buckets was successful.
   NumBuckets = NewBucketCount;
   NumNodes = 0;
 
@@ -356,8 +354,7 @@ bool FoldingSetBase::RemoveNode(Node *N) {
 }
 
 FoldingSetBase::Node *
-FoldingSetBase::GetOrInsertNode(FoldingSetBase::Node *N,
-                                const FoldingSetInfo &Info) {
+FoldingSetBase::GetOrInsertNode(Node *N, const FoldingSetInfo &Info) {
   FoldingSetNodeID ID;
   Info.GetNodeProfile(this, N, ID);
   void *IP;

